content/articles/integrating-instagram-basic-display-api/index.md

```diff
@@ -0,0 +1,919 @@
+
+  ![credentials](/integrating-instagram-basic-display-api/credentials.png)
+
+- After setting your credentials, its time we set up the server in the `src/index.js` file by adding the following:
+
+  ```javascript
+  const express = require("express");
+  const cors = require("cors");
+  require("dotenv").config();
+  const PORT = process.env.PORT || 4000;
+
+  // initialize express
+  const app = express();
+
+  // express configs
+  app.use(cors()); // cors set up
+  app.use(express.json()); // json format
+  app.use(express.urlencoded({ extended: false })); // data parsing
+
+  // routes
+
+  // getting-authorization-code
+  app.get("/get-auth-code", (req, res, next) => {
+    return res.send(
+      `<a href='https://api.instagram.com/oauth/authorize?client_id=${process.env.INSTAGRAM_APP_ID}&redirect_uri=${process.env.REDIRECT_URI}&scope=user_media,user_profile&response_type=code'> Connect to Instagram </a>`
+    );
+  });
+
+  // start server on the PORT.
+  app.listen(PORT, () => console.log(`Server started on port: ${PORT}`));
+  ```
+
+  From above, we are:
+
+  - Requiring the packages we need.
+
+  - Setting the port for our application to run to.
+
+  - Initializing express.
+
+  - Adding some configuration to the express app.
+
+  - Setting up the route for getting an authorization code.
+
+  - Starting the server by listening to the specified port.
+
+- Edit the `scripts` section of `package.json` as follows:
+
+  ![scripts-section](/integrating-instagram-basic-display-api/scripts-section.png)
+
+- In your recently opened VsCode terminal, run the following command:
+
+  ```bash
+  npm run dev
+  ```
+
+  This will start the server on PORT 4000.
+
+- From your web browser, open _http:localhost:4000/get-auth-code_. In there, click the link _Connect to Instagram_ to connect to your Instagram account.
+
+- In the resulting pop-up, click _Allow_, after which you will be redirected to a different page. On this new page, in the URL section, we have a code parameter which is as below:
+
+  ![url-code](/integrating-instagram-basic-display-api/url-code.png)
+
+  Copy the entire code up to where we have `#`. Don't include the `#`. Paste the code to your `.env` file. Your `.env` file should now be similar to:
+
+  ![new-env](/integrating-instagram-basic-display-api/new-env.png)
+
+### Getting the short-lived access token
+
+After getting the authorization code, we can now get the short-lived access token. It is defined as short-lived because it is only valid for one hour. An access token is usually sent along with every request you make to the API for authentication.
+
+To get the short-lived access token, we will follow the following steps:
+
+- Introduce `Apollo server` to our app. To do this, we will edit our `app.js` as follows:
```

Comment:
Up to this point in the article no `app,js` was introduced in the application structure. Kindly define where this file sits in the file structure.

Reply:
Syntax error, it should have been src/index.js. Have solved it.